src/Microsoft.Health.Fhir.Api/Features/Routing/UrlResolver.cs

```diff
@@ -131,18 +131,57 @@ public Uri ResolveRouteUrl(IEnumerable<Tuple<string, string>> unsupportedSearchP
                 {
                     // Remove the parameter if:
                     // 1. It is the continuation token (if there is a new continuation token, then it will be added again).
-                    // 2. The parameter is not supported.
+                    // 2. It is the _sort parameter and the parameter is not supported for sorting
+                    // 3. The parameter is not supported.
                     if (!string.Equals(searchParam.Key, KnownQueryParameterNames.ContinuationToken, StringComparison.OrdinalIgnoreCase))
                     {
-                        IEnumerable<string> removedValues = searchParamsToRemove[searchParam.Key];
+                        if (unsupportedSortingParameters?.Count > 0 && string.Equals(searchParam.Key, KnownQueryParameterNames.Sort, StringComparison.OrdinalIgnoreCase))
+                        {
+                            var filteredValues = new List<string>(searchParam.Value.Count);
+
+                            foreach (string stringValue in searchParam.Value)
+                            {
+                                // parameters are separated by a comma and can be prefixed with a dash to indicate descending order
+                                string[] tokens = stringValue.Split(',');
+
+                                List<string> values = tokens.Where(sortValue =>
+                                        !unsupportedSortingParameters.Any(p =>
+                                            sortValue.EndsWith(p.parameterName, StringComparison.Ordinal) &&
+                                            (sortValue.Length == p.parameterName.Length ||
+                                             (sortValue.Length > 0 && sortValue.Length == p.parameterName.Length + 1 && sortValue[0] == '-'))))
+                                    .ToList();
+
+                                if (values.Count == tokens.Length)
```

Comment:
Is this case needed? It is covered by the two lower cases.

Reply:
We can avoid a string.Join()